### PR TITLE
Chore/circleci windows ccache

### PR DESCRIPTION
**Pull request by CryptoNinjaGeek**
### Scope & Purpose

Get ccache working on our window builds, to cut down compile time.

### CODE REVIEW

**Review by dothebart — APPROVED** [review]
it would be good if there was a way to DRY the docker container names. 
else LGTM.

File: CMakeLists.txt [hunk 1]
```diff
@@ -937,10 +937,52 @@ if (MSVC)
...
+
+      if(CCACHE_RESULT_CODE)
+        set( CCACHE ${CCACHE_REAL} )
+        message(STATUS "Resolved ccache SHIM: '${CCACHE_REAL}'")
+      endif()
+
+      file(COPY_FILE 
+        ${CCACHE} ${CMAKE_BINARY_DIR}/cl.exe
+        ONLY_IF_DIFFERENT
+      )
```
Comment (mpoeter) [inline CMakeLists.txt:962]: I think this deserves a comment about why we need that.

File: .circleci/base_config.yml [hunk 2]
```diff
@@ -344,24 +344,35 @@ jobs:
...
           name: "Build"
           shell: powershell.exe
           command: |
-            docker run --rm --isolation process -v D:/CircleCI/project:C:/project -w C:/project/build arangodb/build-windows-x64:10.0.17763.4737-VS-17.3.3-OpenSSL-3.1.4-f3c77755ece cmake --build . --config `"RelWithDebInfo`" --parallel -- /p:CL_MPcount=24
+            docker run --rm --isolation process -v D:/CircleCI/project:C:/project -v D:/CircleCI-ccache:C:/ccache -w C:/project/build arangodb/build-windows-x64:10.0.17763.4737-VS-17.3.3-OpenSSL-3.1.4-399cc6a6e26 cmake --build . --config `"RelWithDebInfo`" --parallel -- /p:CL_MPcount=24 
+      - run:
+          name: "Print CCache statistics"
+          shell: powershell.exe
+          command: |
+            docker run --rm --isolation process -v D:/CircleCI/project:C:/project -v D:/CircleCI-ccache:C:/ccache -w C:/project/build arangodb/build-windows-x64:10.0.17763.4737-VS-17.3.3-OpenSSL-3.1.4-399cc6a6e26 ccache -s
```
Comment (mpoeter) [inline .circleci/base_config.yml:375]: We probably don't need the project directory, and we also don't care about the work dir
```suggestion
            docker run --rm --isolation process -v D:/CircleCI-ccache:C:/ccache arangodb/build-windows-x64:10.0.17763.4737-VS-17.3.3-OpenSSL-3.1.4-399cc6a6e26 ccache -s
```

**Review by mpoeter — APPROVED** [review]
I think the cmake could use a comment, otherwise LGTM

File: .circleci/base_config.yml (hunk 2 above)
Comment (CryptoNinjaGeek) [reply .circleci/base_config.yml]: Change in multiple places

**Pull request merged**